woocommerce/woocommerce-gutenberg-products-block
# Attribute Filter block: Enable global style

This PR is blocked by #5542

This PR adds support for global style for the `Attribute Filter block`.

Now, the user can edit the style for:
- text-color
- border (color and radius) on the input fields

## Note
Unfortunately at the current stage, global style API doesn't support the customization of the inner element instead of block wrapper, so this PR contains a hacky CSS for accomplishing the goal 

https://github.com/WordPress/gutenberg/issues/33255
https://github.com/WordPress/gutenberg/issues/32417


#4965 

### Screenshots

#### Mode: Display List


![image](https://user-images.githubusercontent.com/4463174/149302585-b909076d-fad3-4304-b4cc-a64c3c8d1d42.png)
*without global style*

![image](https://user-images.githubusercontent.com/4463174/149302246-adc79e1a-78ac-461d-bdbd-85141a8b7ec1.png)
*with global style*


#### Mode: Dropdown List

![image](https://user-images.githubusercontent.com/4463174/149302669-78998d14-d8b7-4109-b6df-fad7997d400b.png)
*without global style*

![image](https://user-images.githubusercontent.com/4463174/149302298-97a885f4-3e79-4fa2-a445-bc038bb53ebf.png)
*with global style*


### Testing

### Manual Testing

Check out this branch:

1.  Upgrade to `WordPress 5.9`.
2. Install and enable the `Twenty Twenty-Two` theme.
3. Add the `Attribute Filter block` to a post.
4. On the right sidebar, personalize the styles of the block.
5. Go on the page and check if there are changes.
6. Reset to default using the `Reset` button from the different sections. 
7. Go to Dashboard and select Appearance > Editor (beta). On top of the screen, select Home > Browser all templates > Single Post. When the page is loaded, add the block to the page.
8. On the Editor page click on the `Styles` icon on the right-top corner.
9. Verify that the `Attribute Filter block` is shown under the `Blocks` section. Personalize again the block.
10. Save your changes.
11. Go on the page created earlier and check if all styles are applied correctly.
12. Edit your previous post/page again.
13. Change again the styles.
14. Save your changes.
15. Check if these styles have priority over the styles from the Site Editor.

### Changelog

> Add support for the global style for the Attribute Filter block.
